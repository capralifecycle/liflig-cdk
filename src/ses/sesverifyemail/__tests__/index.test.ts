import { App, Stack } from "aws-cdk-lib"
import "jest-cdk-snapshot"
import { SesVerifyEmail } from ".."

test("ses-verify-email", () => {
  const app = new App()
  const stack = new Stack(app, "Stack")

  new SesVerifyEmail(stack, "SesVerifyEmail", {
    emailAddress: "example@example.com",
  })

  expect(stack).toMatchCdkSnapshot({
    ignoreAssets: true,
  })
})
