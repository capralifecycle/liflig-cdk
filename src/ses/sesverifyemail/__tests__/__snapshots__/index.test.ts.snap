// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ses-verify-email 1`] = `
Object {
  "Resources": Object {
    "SesVerifyEmail41583370": Object {
      "DeletionPolicy": "Delete",
      "Properties": Object {
        "EmailAddress": "example@example.com",
        "ServiceToken": Object {
          "Fn::GetAtt": Array [
            "lifligcdksesverifyemailproviderProviderframeworkonEventA83D9BF0",
            "Arn",
          ],
        },
      },
      "Type": "AWS::CloudFormation::CustomResource",
      "UpdateReplacePolicy": "Delete",
    },
    "lifligcdksesverifyemailproviderFunction7D5F58AF": Object {
      "DependsOn": Array [
        "lifligcdksesverifyemailproviderFunctionServiceRoleDefaultPolicyF5D78D0B",
        "lifligcdksesverifyemailproviderFunctionServiceRole337D98A1",
      ],
      "Properties": Object {
        "Code": Any<Object>,
        "Handler": "index.handler",
        "Role": Object {
          "Fn::GetAtt": Array [
            "lifligcdksesverifyemailproviderFunctionServiceRole337D98A1",
            "Arn",
          ],
        },
        "Runtime": "nodejs12.x",
        "Timeout": 300,
      },
      "Type": "AWS::Lambda::Function",
    },
    "lifligcdksesverifyemailproviderFunctionServiceRole337D98A1": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "lifligcdksesverifyemailproviderFunctionServiceRoleDefaultPolicyF5D78D0B": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "ses:DeleteIdentity",
                "ses:VerifyEmailIdentity",
              ],
              "Effect": "Allow",
              "Resource": "*",
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "lifligcdksesverifyemailproviderFunctionServiceRoleDefaultPolicyF5D78D0B",
        "Roles": Array [
          Object {
            "Ref": "lifligcdksesverifyemailproviderFunctionServiceRole337D98A1",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "lifligcdksesverifyemailproviderProviderframeworkonEventA83D9BF0": Object {
      "DependsOn": Array [
        "lifligcdksesverifyemailproviderProviderframeworkonEventServiceRoleDefaultPolicyC87308CB",
        "lifligcdksesverifyemailproviderProviderframeworkonEventServiceRoleD5D1F2EA",
      ],
      "Properties": Object {
        "Code": Any<Object>,
        "Description": "AWS CDK resource provider framework - onEvent (Stack/liflig-cdk.ses-verify-email.provider/Provider)",
        "Environment": Object {
          "Variables": Object {
            "USER_ON_EVENT_FUNCTION_ARN": Object {
              "Fn::GetAtt": Array [
                "lifligcdksesverifyemailproviderFunction7D5F58AF",
                "Arn",
              ],
            },
          },
        },
        "Handler": "framework.onEvent",
        "Role": Object {
          "Fn::GetAtt": Array [
            "lifligcdksesverifyemailproviderProviderframeworkonEventServiceRoleD5D1F2EA",
            "Arn",
          ],
        },
        "Runtime": "nodejs14.x",
        "Timeout": 900,
      },
      "Type": "AWS::Lambda::Function",
    },
    "lifligcdksesverifyemailproviderProviderframeworkonEventServiceRoleD5D1F2EA": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "lifligcdksesverifyemailproviderProviderframeworkonEventServiceRoleDefaultPolicyC87308CB": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "lambda:InvokeFunction",
              "Effect": "Allow",
              "Resource": Array [
                Object {
                  "Fn::GetAtt": Array [
                    "lifligcdksesverifyemailproviderFunction7D5F58AF",
                    "Arn",
                  ],
                },
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      Object {
                        "Fn::GetAtt": Array [
                          "lifligcdksesverifyemailproviderFunction7D5F58AF",
                          "Arn",
                        ],
                      },
                      ":*",
                    ],
                  ],
                },
              ],
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "lifligcdksesverifyemailproviderProviderframeworkonEventServiceRoleDefaultPolicyC87308CB",
        "Roles": Array [
          Object {
            "Ref": "lifligcdksesverifyemailproviderProviderframeworkonEventServiceRoleD5D1F2EA",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
  },
}
`;
